agenta-cli/agenta/client/client.py

```diff
@@ -125,18 +127,28 @@ def add_variant_to_server(
         "docker_id": image.docker_id,
         "tags": image.tags,
     }
-    response = requests.post(
-        f"{host}/{BACKEND_URL_SUFFIX}/apps/{app_id}/variant/from-image/",
-        json=payload,
-        headers={"Authorization": api_key} if api_key is not None else None,
-        timeout=600,
-    )
-    if response.status_code != 200:
-        error_message = response.json()
-        raise APIRequestError(
-            f"Request to app_variant endpoint failed with status code {response.status_code} and error message: {error_message}."
-        )
-    return response.json()
+
+    print(f"Waiting for the variant to be ready", end="", flush=True)
+
+    for attempt in range(retries):
+        try:
+            response = requests.post(
+                f"{host}/{BACKEND_URL_SUFFIX}/apps/{app_id}/variant/from-image/",
+                json=payload,
+                headers={"Authorization": api_key} if api_key is not None else None,
+                timeout=600,
+            )
+            response.raise_for_status()
+            print("\nVariant added successfully.")
```

Comment:
Prints should not be part of the client, they are done from the calling function (using colors and nice things :)